File: send_image.py
#!/usr/bin/env python3

import os
import requests
import shutil
from datetime import datetime
from sys import argv
from time import strftime

date = strftime('%d-%m-%Y')
path = '/home/pi/video/'
hours = range(7, 19)
hour = int(strftime('%H'))
days = ('6', '7')
now = datetime.now()
weekday = str(now.isoweekday())
bot_key = os.getenv['MOTION_KEY']


def sends(pict):
    url = f'https://api.telegram.org/{bot_key}/sendPhoto'
    files = {'photo': open(pict, 'rb')}
    data = {'chat_id': "your_chat_id"}
    requests.post(url, files=files, data=data)


def sensor():
    with open('rid.txt') as f:
        motion = f.read()
        if '1' in motion:
            return True


    fls = os.listdir(path)
    to_move = filter(lambda x: x.endswith('.{ext}'.format(ext=ext)), fls)
    path = r'{dir}{date}'.format(dirs=dirs, date=date)

    if not os.path.exists(path):
        os.makedirs(path)

    for obj in to_move:
        shutil.move('{dir}{fls}'.format(fls=obj, dirs=dirs),
                    '{dir}{date}'.format(dir=dir, date=date))


if (hour not in hours or weekday in days) and sensor():
    sends(argv[1])


moveFiles('avi')
moveFiles('jpg')

File: send_image.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"os"
	"time"

	"gopkg.in/telegram-bot-api.v4"
)

var t = time.Now()

func moves() {
	date := t.Format("02-01-2006")
	oldPath := "/oldpath/"
	newPath := "/newpath/" + date + "/"

	fmt.Println(newPath)

	if _, err := os.Stat(newPath); os.IsNotExist(err) {
		os.Mkdir(newPath, 0766)
	}

	files, err := ioutil.ReadDir(oldPath)
	if err != nil {
		log.Fatal(err)
	}
	for _, file := range files {
		if !file.IsDir() {
			err := os.Rename(oldPath+file.Name(), newPath+file.Name())
			if err != nil {
				panic(err)
			}
			fmt.Println(file.Name())
		}
	}
}

func checks() bool {
	hour := t.Hour()
	weekday := t.Weekday()
	if hour <= 8 && hour >= 19 || weekday == time.Sunday || weekday == time.Saturday {
		return true
	}
	return false
}

func main() {
	bot, err := tgbotapi.NewBotAPI("247646851:AAH4GQny2X_iMhvbbdYYFRUFc5AslBkEfkc")
	if err != nil {
		log.Panic(err)
	}

	photo := tgbotapi.NewPhotoUpload(2057901, os.Args[1])
	if checks() {
		bot.Send(photo)
	} else {
		fmt.Println("range is out")
	}

	moves()
}

// def sensor():
//     fl = open('path_to_rid')
//     motion = fl.read()
//     if '1' in motion:
//         return True
